client/src/pages/MentionPage.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getOneMention } from "../hooks/getOneMention";
+import HomePage from "./HomePage";
+import {
+  Dialog,
+  Typography,
+  Grid,
+  Button,
+  Link,
+  Card,
+  CardMedia,
+  Box,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import { REACT_APP_BASE_URL } from "../utils/constants";
+import redditLogo from "../utils/images/reddit-logo.png";
+
+const useStyles = makeStyles((theme) => ({
+  modalRoot: {
+    width: "100%",
+    height: "600px",
+  },
+  modalHeader: {
+    minHeight: "5em",
+    alignItems: "center",
+    margin: "1em 2em",
+  },
+  modalImage: {
+    height: "16em",
+    width: "16em",
+    margin: "auto",
+  },
+  modalContent: {
+    height: "16em",
+    width: "100%",
+    overflow: "auto",
+  },
+  modalButtons: {
+    margin: "0.5em",
+  },
+}));
+
+const MentionPage = () => {
+  const { id } = useParams();
+  const classes = useStyles();
+  const [mention, setMention] = useState({});
+  const url = `${REACT_APP_BASE_URL}/mentions/${mention._id}`;
+  const [mentionUrl, setMentionUrl] = useState(url);
+  const [openDialog, setOpenDialog] = useState(false);
+
+  useEffect(() => {
+    getOneMention(id)
+      .then((data) => {
+        setMention(data);
+        setOpenDialog(true);
+      })
+      .catch((err) => alert("Something went wrong"));
+  }, []);
+
+  const handleclick = () => {
+    setOpenDialog(false);
+  };
```

Comment:
for MentionPage.js, should the url change when you close the modal?

Reply:
Well spotted ! I took care of it